scala-lang/src/test/scala/com/baeldung/scala/callbynameandvalue/CallByNameCallByValueTest.scala

```diff
@@ -0,0 +1,34 @@
+package com.baeldung.scala.callbynameandvalue
+
+import com.baeldung.scala.callbynameandvalue.CallByNameCallByValue._
+import org.scalatest.{FlatSpec, Matchers}
+
+/**
+ * @author vid2010
+ */
+class CallByNameCallByValueTest extends FlatSpec with Matchers {
+
+  "A call by-value" should "have evaluated before passing an argument" in {
+    getTimeByVal(System.currentTimeMillis()) should be(System.currentTimeMillis())
```

Comment:
It might be a good idea to define the time as a variable to avoid the possibility of a race condition. For example:
`val currentTime = System.currentTimeMillis()
getTimeByVal(currentTime) should be(currentTime)`

Reply:
Thank you @glmartin ,
I have added the changes.